### PR TITLE
add deepmerge v4.x.x defs

### PR DESCRIPTION
**Pull request by magicmark**
- Link to GitHub or NPM: https://www.npmjs.com/package/deepmerge
- Type of contribution: new definition

Other notes:
- Copied tests from https://github.com/flow-typed/flow-typed/tree/master/definitions/npm/deepmerge_v2.1.x

### CODE REVIEW

File: definitions/npm/deepmerge_v4.x.x/flow_v0.100.x-/deepmerge_v4.x.x.js [hunk 1]
```diff
@@ -0,0 +1,16 @@
+// Converted from https://github.com/TehShrike/deepmerge/blob/ad8cfddaa8f4b79f5e2c6943445a8b42377ec495/index.d.ts#L4
+
+declare module 'deepmerge' {
+    declare type Options = {
```
Comment (pascalduez) [inline definitions/npm/deepmerge_v4.x.x/flow_v0.100.x-/deepmerge_v4.x.x.js]: Needs to be either exact `{| |}` or inexact `{ ... }`.

File: definitions/npm/deepmerge_v4.x.x/flow_v0.100.x-/deepmerge_v4.x.x.js [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+        arrayMerge?: (target: any[], source: any[], options?: Options) => Array<any>,
+        clone?: boolean,
+        customMerge?: (key: string, options?: Options) => ?(x: any, y: any) => any,
+        isMergeableObject?: (value: Object) => boolean,
```
Comment (pascalduez) [inline definitions/npm/deepmerge_v4.x.x/flow_v0.100.x-/deepmerge_v4.x.x.js]: `Object` is deprecated, if you can't gain a more precise type, then `{ ... }` should be used.

File: definitions/npm/deepmerge_v4.x.x/flow_v0.100.x-/test_deepmerge_v4.x.x.js [hunk 3]
```diff
@@ -0,0 +1,8 @@
+const deepmerge = require("deepmerge");
+
+const m1: { a: string, b: string, ... } = deepmerge({a: ''}, { b: '' })
```
Comment (pascalduez) [inline definitions/npm/deepmerge_v4.x.x/flow_v0.100.x-/test_deepmerge_v4.x.x.js]: Tests needs to be [wrapped with `describe / it` blocks](https://github.com/flow-typed/flow-typed/blob/master/CONTRIBUTING.md#use-describe-and-it-blocks-to-limit-scope).

**Comment by GAntoine** [comment]
Closing since stale

**Pull request closed without merging**